packages/@aws-cdk/aws-servicecatalogappregistry/lib/application.ts

```diff
@@ -275,10 +275,8 @@ export class Application extends ApplicationBase {
     this.applicationName = props.applicationName;
     this.nodeAddress = cdk.Names.nodeUniqueId(application.node);
 
-    this.applicationManagerUrl = new cdk.CfnOutput(this, 'ApplicationManagerUrl', {
-      value: `https://${this.env.region}.console.aws.amazon.com/systems-manager/appmanager/application/AWS_AppRegistry_Application-${this.applicationName}`,
-      description: 'Application manager url for the application created.',
-    });
+    this.applicationManagerUrl =
+      `https://${this.env.region}.console.aws.amazon.com/systems-manager/appmanager/application/AWS_AppRegistry_Application-${this.applicationName}`;
```

Comment:
This defeats the purpose of providing easier access to the application URL when the application is created through `TargetApplication.createApplicationStack`. 

A new pull request #24386 was create to expose this URL as CFN output on the stack defined by `TargetApplication.createApplicationStack` construct, where the application is created for later association. That got closed as duplicate.

Reply:
It seems like this original design choice was not really the right choice here. Sure, it's easier for certain use cases to make this a `CfnOutput`, but it seems that there are use cases to do otherwise as well.

Reply:
`TargetApplication` & `ApplicationAssociator` are intended to be included into default cdk template through `cdk init`, therefore all the cdk users can easily use the feature: all the stacks deployed by the cdk package can be contained in one application .

`TargetApplication.createApplicationStack` creates a new stack, defined by [aws-servicecatalogappregistry](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-servicecatalogappregistry/lib/target-application.ts#L101), in customers account where [an application is created](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-servicecatalogappregistry/lib/target-application.ts#L102).

For the customers who are onboard with [`ApplicationAssociator`](https://github.com/aws/aws-cdk/tree/main/packages/%40aws-cdk/aws-servicecatalogappregistry#application-associator) construct, the service team wants to provide an easier console access to the application created implicitly through `TargetApplication.createApplicationStack`. Therefore customers can view the application and its associated stacks without further cdk code change. Therefore the service team chose to expose the URL as CFN output in the stack, so that the console access to the application can be easily obtained. Since the CFN output stays in the stack fully managed by the `aws-servicecatalogappregistry` constructs, it was considered to be minimal customer impacting.

Is there any other suggestion to expose the URL while having minimal customer intervention?

Reply:
I think there is a middle ground we can take:
- Make `Application` construct having applicationManagerUrl as string type so that customers can use it to create CFN Output as needed.
- In `TargetApplication` construct, `createApplicationStack` will use `applicationManagerUrl` property to create the CFN Output in the Stack created by the CDK construct.

Reply:
> Is there any other suggestion to expose the URL while having minimal customer intervention?

You can always make a boolean option: `emitConsoleUrlAsOutput?: boolean`.

Reply:
Your proposal seems like a good middle ground by the way.

But you will always have to contend with customer opinion, which means making it configurable. I don't mind the default being `true`, as long as it can be switched off for those who care.

Reply:
Thanks for your comments, new PR is published: https://github.com/aws/aws-cdk/pull/24483